aiohttp/pytest_plugin.py

```diff
@@ -55,7 +56,10 @@ def test_client(loop):
 
     @asyncio.coroutine
     def _create_from_app_factory(app_factory, *args, **kwargs):
-        app = app_factory(loop, *args, **kwargs)
+        if not isinstance(app_factory, Application):
+            app = app_factory(loop, *args, **kwargs)
+        else:
+            app = app_factory
```

Comment:
I mean 
```
if isinstance(app_factory, Application):
    assert not args, "arg should be empty"
    assert not kwargs, "kwargs should be empty"
    ....
```
It will prevent cases like: `yield from test_client(app, 1, 2, param=3)`.